pkg/sql/types/types.go

```diff
@@ -185,7 +185,8 @@ type T struct {
 // UserDefinedTypeMetadata contains metadata needed for runtime
 // operations on user defined types. The metadata must be read only.
 type UserDefinedTypeMetadata struct {
-	Name UserDefinedTypeName
+	// Name is the resolved name of this type.
+	Name *UserDefinedTypeName
```

Comment:
can Name ever be nil?

Reply:
I guess if it wasn't hydrated it'd be nil, right?

Reply:
Yeah, it can be nil if the type isn't hydrated. The interface before could be nil, so the code is already aware of the possibility of it being nil.